streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStore.java

```diff
@@ -119,6 +128,10 @@ public void remove(final Bytes rawBaseKey) {
 
     abstract protected KeyValue<Bytes, byte[]> getIndexKeyValue(final Bytes baseKey, final byte[] baseValue);
 
+    protected long getActualFrom(final long from) {
+        return Math.max(from, observedStreamTime - retentionPeriod + 1);
```

Comment:
I think we allow `retentionPeriod == 0` and thus, I am wondering if `+ 1` is correct here?